### ISSUE TITLE
bug: `SumOp.backward` fails for `keepdim=False` with non-leading reduced axis

### ISSUE THREAD

**Issue by az-fouche** [az-fouche/nanotorch#37]
`src/nanotorch/autograd/ops.py:327`:

```python
return (grad_out.expand(self._orig_shape),)
```

For `x.shape == (3, 4, 5)` with `sum(axis=1, keepdim=False)`:
- `grad_out.shape == (3, 5)`
- `orig_shape == (3, 4, 5)`
- `expand` (`core.py:385-401`) right-aligns: position -1 sees `5 vs 5` ✓, position -2 sees `3 vs 4` → raises.

Currently masked because all MLP reductions are over the batch axis (leading) or `axis=None` (grad_out is rank-0).

Fix: reshape `grad_out` to the keepdim shape (insert 1s at the dropped axes) before expand. Same fix needed in `MeanOp.backward` (`:370`).

Will trip the moment we do cross-entropy / per-feature reductions / softmax.
